urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/Game_World/GameEngine.java

```diff
@@ -0,0 +1,129 @@
+package Game_World;
+
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.beans.IntrospectionException;
+import java.io.FileNotFoundException;
+import java.io.Serializable;
+import java.util.HashSet;
+
+import Data_Storage.XMLReader;
+import Data_Storage.XMLWriter;
+import Game_World_Objects.Chest;
+import Game_World_Objects.Key;
+import Junit_Tests.Class_Examples;
+import Object_Interfaces.GameObject;
+
+public class GameEngine implements Serializable {
+	/**
+	 * The GameWorld loading engine.
+	 * This will act as the interface to the whole Game_World package, loading a GameWorld from
+	 * an XML file or using a constructor.
+	 * @author Danesh Abeyratne
+	 *
+	 */
+
+	private String xmlFile;
+	private int level = 1;
+	private GameWorld world;
+	private HashSet<Player> players;
+
+	public GameEngine(){
+
+		if(level == 1){
+			//in = (FileInputStream) this.getClass().getClassLoader().getResourceAsStream("/GameWorld_Resources/level1test.xml");
+			xmlFile = "levelOne.xml"; //"src/GameWorld_Resources/levelOne.xml";
+			//xmlFile = "src/GameWorld_Resources/level1test.xml";
+		}
+		else if(level == 2){
+			xmlFile = "level2.xml";
+		}
+
+		XMLReader reader;
+		try {
+			reader = new XMLReader(xmlFile);
+			world = reader.readGameWorld();
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 43:48, 56:61.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.